allennlp/common/params.py

```diff
@@ -211,6 +211,22 @@ def log_recursively(parameters, history):
         log_recursively(self.params, self.history)
         return self.params
 
+    def as_flat_dict(self):
+        """
+        Returns the parameters of a flat dictionary from keys to values.
+        Nested structure is collapsed with periods.
+        """
+        flat_params = {}
+        def recurse(parameters, path):
+            for key, value in parameters.items():
+                if isinstance(value, dict):
+                    recurse(value, path + [key])
+                else:
+                    flat_params['.'.join(path + [key])] = value
```

Comment:
nit: you use the same logic in two places, might be clearer if you just did

```
newpath = path + [key]
```

once